File: Java/2264.java
class Solution {
    public String largestGoodInteger(String num) {
        int ans = -1;
        for (int i = 0; i + 2 < num.length(); i++) {
            if (num.charAt(i) == num.charAt(i + 1) && num.charAt(i) == num.charAt(i + 2))
                ans = Math.max(ans, num.charAt(i) - '0');
        }
        StringBuilder sb = new StringBuilder(3);
        for (int i = 0; i < 3; i++)
            sb.append((char) (48 + ans));
        return ans != -1 ? sb.toString() : "";
    }
}

File: Python/2264.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        result = ""
        for i in range(2, len(num)):
            if num[i - 2] == num[i - 1] == num[i]:
                result = max(result, num[i] * 3)
        return result
